index/server/main.go

```diff
@@ -361,7 +445,38 @@ func ociServerProxy(c *gin.Context) {
 	proxy := httputil.NewSingleHostReverseProxy(remote)
 
 	// Set up the request to the proxy
-	// This is a good place to set up telemetry for requests to the OCI server (e.g. by parsing the path)
+	// Track event for telemetry
+	if enableTelemetry {
+		proxyPath := c.Param("proxyPath")
+		if proxyPath != "" {
+			var name string
+			var resource string
+			parts := strings.Split(proxyPath, "/")
+			if len(parts) == 5 {
```

Comment:
Why do we need to check for both lengths 4 and 5? Won't the proxyPath we want always be something like `/devfile-catalog/java-quarkus/blobs/sha256:d913cab108c3bc1bd06ce61f1e0cdb6eea2222a7884378f7e656fa26249990b9`, so the length we want would be 4? (`devfile-catalog`, `java-quarkus`, `blobs`, and `sha:...` in this example)

Reply:
Currently it always be `/v2/devfile-catalog/nodejs/blobs/sha256:cc3bf36be550d38b701753da58034143da93ec387693f4ef98136e3e06e71fe1` (prepend API version), but I think you brought up we might receive `/devfile-catalog/*` (without API version) on previous code review, that's why I make the splitting part more robust to support the two APIs (check both lengths 4 and 5)

Reply:
ProxyPath is the path that follows /v2, so I don’t think it will ever contain /v2 in it. So I just think we just need to check for an array of length 4

Reply:
You're right, the proxyPath doesn't contain the version, but the only length we need to check is 5 instead of 4. The reason is we use `split("/")` so the first element would be `""` which means for this line `parts := strings.Split(proxyPath, "/")`, if the proxyPath is `/devfile-catalog/java-quarkus/blobs/sha256:d913cab108c3bc1bd06ce61f1e0cdb6eea2222a7884378f7e656fa26249990b9` the parts array would be:

```
array[0]: ""
array[1]: devfile-catalog
array[2]: java-quarkus
array[3]: blobs
array[4]: sha256:d913cab108c3bc1bd06ce61f1e0cdb6eea2222a7884378f7e656fa26249990b9
```

I also test it by injecting some debug logs below
```
2021/08/23 15:44:45 proxyPath is /devfile-catalog/nodejs/blobs/sha256:cc3bf36be550d38b701753da58034143da93ec387693f4ef98136e3e06e71fe1
2021/08/23 15:44:45 parts len is 5
2021/08/23 15:44:45 name is nodejs, resource is blobs
```

with source code
```
                        parts := strings.Split(proxyPath, "/")
			log.Printf("parts len is %d\n", len(parts))
			// Check proxyPath with API version - "/v2/devfile-catalog/nodejs/blobs/sha256:cc3bf36be550d38b701753da58034143da93ec387693f4ef98136e3e06e71fe1"
			if len(parts) == 5 {
				name = parts[2]
				resource = parts[3]
				log.Printf("name is %s, resource is %s\n", name, resource)
			} else if len(parts) == 4 {
				// Check proxyPath without API version - "/devfile-catalog/java-quarkus/blobs/sha256:d913cab108c3bc1bd06ce61f1e0cdb6eea2222a7884378f7e656fa26249990b9"
				name = parts[1]
				resource = parts[2]
				log.Printf("name is %s, resource is %s\n", name, resource)
			}
```

Reply:
Updated

Reply:
🤦‍♂️ You're right, I missed the empty space when I was testing.